client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java

```diff
@@ -131,7 +131,7 @@ private static class ReduceFileGroups {
   // key: shuffleId
   private final Map<Integer, ReduceFileGroups> reduceFileGroupsMap = new ConcurrentHashMap<>();
 
-  private TransportClient currentClient;
+  private ConcurrentHashMap<String, TransportClient> currentClient = new ConcurrentHashMap<>();
```

Comment:
need cleanup this when the map task/shuffle is finished.

Reply:
yes, it should be cleauped in cleaup function